vector/src/main/java/im/vector/fragments/VectorPublicRoomsListFragment.java

```diff
@@ -139,31 +140,30 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
                 if (null != publicRoom.roomId) {
                     final RoomPreviewData roomPreviewData = new RoomPreviewData(mSession, publicRoom.roomId, null, publicRoom.getAlias(), null);
 
+                    // Check whether the room exists to handled the cases where the user is invited or he has joined.
+                    // CAUTION: the room may exist whereas the user membership is neither invited nor joined.
                     Room room = mSession.getDataHandler().getRoom(publicRoom.roomId, false);
+                    if (null != room && room.hasMembership(RoomMember.MEMBERSHIP_INVITE)) {
```

Comment:
This block of code seems very close to the other ones. Refactor if possible ?

Reply:
I agree with you, but I will not be able to refactor it now :(